storage/grpc_client.go

```diff
@@ -1359,16 +1357,12 @@ func (c *grpcStorageClient) NewMultiRangeDownloader(ctx context.Context, params
 		mrd.mu.Unlock()
 	}
 
-	mrd.mu.Lock()
```

Comment:
IIUC, this is checking for a deadlock. Why don’t we need it anymore?

Reply:
It was added as a precaution on any uses of any field in the `gRPCBidiReader` struct, because that struct is accessed/modified concurrently. 

However, the field this mutex is protecting here, `mrd.objectSize`, is only ever used to create the `Attrs` struct we return to the user. So, I removed the `objectSize` field from the `gRPCBidiReader` struct, and I moved `obj.GetSize()` to where it is used. Similar to the other field of `Attrs`, it does not need concurrency protection as we return it immediately and never use it again.